### PR TITLE
L10n cleanup and xliff+new translations

### PR DESCRIPTION
**Pull request by kenmanheimer**
Commission four more languages: French, Portuguese, German, and Russian.

> NOTE: This PR includes the l10n-cleanup-and-xliff branch on which it depends. It's probably best to pull that one first: #633.

Also new revision of Spanish.

I've poked around and confirmed basic soundness with everything but Russian, which isn't easily present on the emulator.
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/SpiderOak/SpiderOakMobileClient/pull/636%23discussion_r23502006%22%2C%20%22https%3A//github.com/SpiderOak/SpiderOakMobileClient/pull/636%23discussion_r23502012%22%2C%20%22https%3A//github.com/SpiderOak/SpiderOakMobileClient/pull/636%23discussion_r23502013%22%2C%20%22https%3A//github.com/SpiderOak/SpiderOakMobileClient/pull/636%23discussion_r23502015%22%2C%20%22https%3A//github.com/SpiderOak/SpiderOakMobileClient/pull/636%23discussion_r23502016%22%2C%20%22https%3A//github.com/SpiderOak/SpiderOakMobileClient/pull/636%23discussion_r23502017%22%2C%20%22https%3A//github.com/SpiderOak/SpiderOakMobileClient/pull/636%23issuecomment-71353552%22%2C%20%22https%3A//github.com/SpiderOak/SpiderOakMobileClient/pull/636%23issuecomment-71536151%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/SpiderOak/SpiderOakMobileClient/pull/636%23issuecomment-71353552%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%40kenmanheimer%20-%20You%20might%20wanna%20go%20over%20these%20translations%20a%20bit...%20I%20am%20finding%20%27Zero-Knowledge%27%20translated%20quite%20frequently%20%28I%20thought%20we%20were%20leaving%20it%29%2C%20and%20in%20some%20places%20it%20is%20in%20the%20app%20as%20%27%5B%5BZero-Knowledge%5D%5D%27%20complete%20with%20%60%5B%5B%60%27s%20%28such%20as%20this%20screen%20in%20%60pt%60%3A%20http%3A//cl.ly/image/0W0D1h2L0v3m%29%5Cr%5Cn%5Cr%5Cn%22%2C%20%22created_at%22%3A%20%222015-01-25T05%3A29%3A28Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/554999%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/devgeeks%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40devgeeks%2C%20excellent%20catches.%20I%27ve%20done%20a%20visual%20inspection%20of%20the%20json%20l10n%20string%20files%20%28using%20regexp%20isearch%20to%20highlight%20the%20bracketed%20strings%2C%20etc%29%20to%20look%20for%20the%20ones%20you%20found%20and%20any%20others%20that%20might%20be%20lurking.%20It%20looks%20like%20I%20got%20all%20the%20ones%20you%20caught%20%28evidenced%20by%20the%20commented%20diffs%20being%20outdated%29%2C%20and%20may%20have%20gotten%20a%20few%20more.%20I%20definitely%20found%20a%20string%20that%20I%20neglected%20to%20bracket%20-%20remedied%20-%20plus%20a%20few%20plain%20strings%20in%20the%20Russian%20translation%20that%20were%20not%20missed%20in%20the%20translation.%5Cr%5Cn%5Cr%5CnI%20was%20exact%20about%20tallying%20the%20changes%20to%20be%20able%20to%20send%20the%20specifics%20to%20the%20translators%2C%20since%20we%27re%20just%20at%20the%20verge%20of%20using%20xliff%20to%20automatically%20keep%20track%20of%20that%20sort%20of%20thing.%5Cr%5Cn%5Cr%5CnI%27ve%20confirmed%20that%20this%20works%20fine%20in%20a%20few%20languages%2C%20but%20not%20all.%22%2C%20%22created_at%22%3A%20%222015-01-26T20%3A54%3A12Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/515685%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kenmanheimer%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%204d20da5dc1c8b2a5f72844ca81bd0148dfaf07bf%20www/locales/de.json%20146%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/SpiderOak/SpiderOakMobileClient/pull/636%23discussion_r23502006%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Looks%20like%20this%20is%20missing%20a%20%60%7B%60%22%2C%20%22created_at%22%3A%20%222015-01-25T05%3A21%3A14Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/554999%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/devgeeks%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20www/locales/de.json%3AL1-240%22%7D%2C%20%22Pull%204d20da5dc1c8b2a5f72844ca81bd0148dfaf07bf%20xliff/ru.xml%20570%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/SpiderOak/SpiderOakMobileClient/pull/636%23discussion_r23502017%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Is%20Zero-Knowledge%20supposed%20to%20be%20translated%20here%3F%22%2C%20%22created_at%22%3A%20%222015-01-25T05%3A26%3A35Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/554999%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/devgeeks%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20xliff/ru.xml%3AL1-960%22%7D%2C%20%22Pull%204d20da5dc1c8b2a5f72844ca81bd0148dfaf07bf%20xliff/ru.xml%20562%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/SpiderOak/SpiderOakMobileClient/pull/636%23discussion_r23502016%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Is%20Zero-Knowledge%20supposed%20to%20be%20translated%20here%3F%22%2C%20%22created_at%22%3A%20%222015-01-25T05%3A26%3A23Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/554999%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/devgeeks%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20xliff/ru.xml%3AL1-960%22%7D%2C%20%22Pull%204d20da5dc1c8b2a5f72844ca81bd0148dfaf07bf%20xliff/pt.xml%20562%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/SpiderOak/SpiderOakMobileClient/pull/636%23discussion_r23502013%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Is%20Zero-Knowledge%20supposed%20to%20be%20translated%20here%3F%22%2C%20%22created_at%22%3A%20%222015-01-25T05%3A25%3A47Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/554999%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/devgeeks%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20xliff/pt.xml%3AL1-960%22%7D%2C%20%22Pull%204d20da5dc1c8b2a5f72844ca81bd0148dfaf07bf%20xliff/fr.xml%20562%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/SpiderOak/SpiderOakMobileClient/pull/636%23discussion_r23502012%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Is%20Zero-Knowledge%20supposed%20to%20be%20translated%20here%3F%22%2C%20%22created_at%22%3A%20%222015-01-25T05%3A25%3A19Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/554999%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/devgeeks%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20xliff/fr.xml%3AL1-960%22%7D%2C%20%22Pull%204d20da5dc1c8b2a5f72844ca81bd0148dfaf07bf%20xliff/pt.xml%20570%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/SpiderOak/SpiderOakMobileClient/pull/636%23discussion_r23502015%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Is%20Zero-Knowledge%20supposed%20to%20be%20translated%20here%3F%22%2C%20%22created_at%22%3A%20%222015-01-25T05%3A26%3A03Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/554999%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/devgeeks%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20xliff/pt.xml%3AL1-960%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [x] <a href='#crh-comment-Pull 4d20da5dc1c8b2a5f72844ca81bd0148dfaf07bf www/locales/de.json 146'></a> &nbsp;<b><a href='https://github.com/SpiderOak/SpiderOakMobileClient/pull/636#discussion_r23502006'>File: www/locales/de.json:L1-240</a></b>
- <a href='https://github.com/devgeeks'></a> Looks like this is missing a `{`
- [x] <a href='#crh-comment-Pull 4d20da5dc1c8b2a5f72844ca81bd0148dfaf07bf xliff/fr.xml 562'></a> &nbsp;<b><a href='https://github.com/SpiderOak/SpiderOakMobileClient/pull/636#discussion_r23502012'>File: xliff/fr.xml:L1-960</a></b>
- <a href='https://github.com/devgeeks'></a> Is Zero-Knowledge supposed to be translated here?
- [x] <a href='#crh-comment-Pull 4d20da5dc1c8b2a5f72844ca81bd0148dfaf07bf xliff/pt.xml 562'></a> &nbsp;<b><a href='https://github.com/SpiderOak/SpiderOakMobileClient/pull/636#discussion_r23502013'>File: xliff/pt.xml:L1-960</a></b>
- <a href='https://github.com/devgeeks'></a> Is Zero-Knowledge supposed to be translated here?
- [x] <a href='#crh-comment-Pull 4d20da5dc1c8b2a5f72844ca81bd0148dfaf07bf xliff/pt.xml 570'></a> &nbsp;<b><a href='https://github.com/SpiderOak/SpiderOakMobileClient/pull/636#discussion_r23502015'>File: xliff/pt.xml:L1-960</a></b>
- <a href='https://github.com/devgeeks'></a> Is Zero-Knowledge supposed to be translated here?
- [x] <a href='#crh-comment-Pull 4d20da5dc1c8b2a5f72844ca81bd0148dfaf07bf xliff/ru.xml 562'></a> &nbsp;<b><a href='https://github.com/SpiderOak/SpiderOakMobileClient/pull/636#discussion_r23502016'>File: xliff/ru.xml:L1-960</a></b>
- <a href='https://github.com/devgeeks'></a> Is Zero-Knowledge supposed to be translated here?
- [x] <a href='#crh-comment-Pull 4d20da5dc1c8b2a5f72844ca81bd0148dfaf07bf xliff/ru.xml 570'></a> &nbsp;<b><a href='https://github.com/SpiderOak/SpiderOakMobileClient/pull/636#discussion_r23502017'>File: xliff/ru.xml:L1-960</a></b>
- <a href='https://github.com/devgeeks'></a> Is Zero-Knowledge supposed to be translated here?
- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/SpiderOak/SpiderOakMobileClient/pull/636#issuecomment-71353552'>General Comment</a></b>
- <a href='https://github.com/devgeeks'></a> @kenmanheimer - You might wanna go over these translations a bit... I am finding 'Zero-Knowledge' translated quite frequently (I thought we were leaving it), and in some places it is in the app as '[[Zero-Knowledge]]' complete with `[[`'s (such as this screen in `pt`: http://cl.ly/image/0W0D1h2L0v3m)
- <a href='https://github.com/kenmanheimer'></a> @devgeeks, excellent catches. I've done a visual inspection of the json l10n string files (using regexp isearch to highlight the bracketed strings, etc) to look for the ones you found and any others that might be lurking. It looks like I got all the ones you caught (evidenced by the commented diffs being outdated), and may have gotten a few more. I definitely found a string that I neglected to bracket - remedied - plus a few plain strings in the Russian translation that were not missed in the translation.
I was exact about tallying the changes to be able to send the specifics to the translators, since we're just at the verge of using xliff to automatically keep track of that sort of thing.
I've confirmed that this works fine in a few languages, but not all.

<a href='https://www.codereviewhub.com/SpiderOak/SpiderOakMobileClient/pull/636?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/SpiderOak/SpiderOakMobileClient/pull/636?approve=1'></a>&nbsp;<a href='https://github.com/SpiderOak/SpiderOakMobileClient/pull/636'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by kenmanheimer** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/SpiderOak/SpiderOakMobileClient/pull/636'>Refresh</a>.<a href='#crh-update-none'></a>

File: www/locales/de.json [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+	"If you prefer to use only '[[Zero-Knowledge]]' access, cancel this login, and only login via the [[SpiderOak]] desktop client.": "Wenn Sie ausnahmslos den Zugang nach dem Prinzip  ‚{{Zero-Knowledge}}' verwenden möchten, brechen Sie diese Anmeldung ab und melden Sie sich nur über den {{SpiderOak}}-Desktop-Client an.",
+	"Continue logging in": "Anmeldung fortsetzen",
+	"Don't login now": "Jetzt nicht anmelden",
+	"Passcode unlock": "Entsperren des Passcodes",
+	"Learn more about SpiderOak &raquo;": "Weitere Infos zu {SpiderOak}}&raquo;",
```
Comment (devgeeks) [inline www/locales/de.json]: Looks like this is missing a `{`

File: xliff/fr.xml [hunk 2]
```diff
@@ -0,0 +1,959 @@
...
+        <source>Live Privately,</source>
+        <target>Vivez en toute confidentialité,</target>
+      </trans-unit>
+      <trans-unit id="138">
+        <source>The SpiderOak Team</source>
+        <target>L'équipe SpiderOak</target>
+      </trans-unit>
+      <trans-unit id="139">
+        <source>Not '{{Zero-Knowledge}}'</source>
+        <target>Pas '{{ Communication d'information zéro }}'</target>
```
Comment (devgeeks) [inline xliff/fr.xml]: Is Zero-Knowledge supposed to be translated here?

File: xliff/pt.xml [hunk 3]
```diff
@@ -0,0 +1,959 @@
...
+        <source>Live Privately,</source>
+        <target>Viva com Privacidade,</target>
+      </trans-unit>
+      <trans-unit id="138">
+        <source>The SpiderOak Team</source>
+        <target>A Equipe SpiderOak</target>
+      </trans-unit>
+      <trans-unit id="139">
+        <source>Not '[[Zero-Knowledge]]'</source>
+        <target>Não é '{{Conhecimento-Zero}}'</target>
```
Comment (devgeeks) [inline xliff/pt.xml]: Is Zero-Knowledge supposed to be translated here?

File: xliff/pt.xml [hunk 4]
```diff
@@ -0,0 +1,959 @@
...
+      </trans-unit>
+      <trans-unit id="140">
+        <source>If you login with this mobile client, or via the website, your password will exist in the [[SpiderOak]] server memory during your browsing session. However, your password will be stored in encrypted memory and never written to an unencrypted disk. When your session is done the server's copy of your password is completely erased.</source>
+        <target>Se você entrar com este cliente móvel, ou através do website, a sua senha estará na memória do servidor {{SpiderOak}} durante a sua sessão de navegação. No entanto, a sua senha será armazenada em memória criptografada e jamais será guardada em um disco não-criptografado. Quando a sua sessão terminar, a cópia da sua senha será completamente eliminada do servidor.</target>
+      </trans-unit>
+      <trans-unit id="141">
+        <source>If you prefer to use only '[[Zero-Knowledge]]' access, cancel this login, and only login via the [[SpiderOak]] desktop client.</source>
+        <target>Se você prefere utilizar somente o acesso com '{{Conhecimento-Zero}}', cancele este login e entre apenas através do cliente {{SpiderOak}} no seu computador.</target>
```
Comment (devgeeks) [inline xliff/pt.xml]: Is Zero-Knowledge supposed to be translated here?

File: xliff/ru.xml [hunk 5]
```diff
@@ -0,0 +1,959 @@
...
+        <source>Live Privately,</source>
+        <target>Сохраняйте конфиденциальность</target>
+      </trans-unit>
+      <trans-unit id="138">
+        <source>The SpiderOak Team</source>
+        <target>Группа SpiderOak</target>
+      </trans-unit>
+      <trans-unit id="139">
+        <source>Not '{{Zero-Knowledge}}'</source>
+        <target>Не '{{нулевое разглашение}}'</target>
```
Comment (devgeeks) [inline xliff/ru.xml]: Is Zero-Knowledge supposed to be translated here?

File: xliff/ru.xml [hunk 6]
```diff
@@ -0,0 +1,959 @@
...
+      </trans-unit>
+      <trans-unit id="140">
+        <source>If you login with this mobile client, or via the website, your password will exist in the [[SpiderOak]] server memory during your browsing session. However, your password will be stored in encrypted memory and never written to an unencrypted disk. When your session is done the server's copy of your password is completely erased.</source>
+        <target>Если вы заходите с помощью этого мобильного клиента или через веб-сайт, ваш пароль будет оставаться в памяти сервера {{SpiderOak}} в течение сеанса браузинга. Однако ваш пароль будет храниться в зашифрованной памяти и не будет записываться на незашифрованный диск.  После завершения сеанса, копия вашего пароля на сервере полностью стирается.</target>
+      </trans-unit>
+      <trans-unit id="141">
+        <source>If you prefer to use only '{{Zero-Knowledge}}' access, cancel this login, and only login via the [[SpiderOak]] desktop client.</source>
+        <target>Если вы предпочитаете использовать только доступ с '{{нулевое разглашение}}', отмените этот логин и заходите в систему только через клиента для настольных ПК {{SpiderOak}}.</target>
```
Comment (devgeeks) [inline xliff/ru.xml]: Is Zero-Knowledge supposed to be translated here?

**Comment by devgeeks** [comment]
@kenmanheimer - You might wanna go over these translations a bit... I am finding 'Zero-Knowledge' translated quite frequently (I thought we were leaving it), and in some places it is in the app as '[[Zero-Knowledge]]' complete with `[[`'s (such as this screen in `pt`: http://cl.ly/image/0W0D1h2L0v3m)

**Comment by kenmanheimer** [comment]
@devgeeks, excellent catches. I've done a visual inspection of the json l10n string files (using regexp isearch to highlight the bracketed strings, etc) to look for the ones you found and any others that might be lurking. It looks like I got all the ones you caught (evidenced by the commented diffs being outdated), and may have gotten a few more. I definitely found a string that I neglected to bracket - remedied - plus a few plain strings in the Russian translation that were not missed in the translation.

I was exact about tallying the changes to be able to send the specifics to the translators, since we're just at the verge of using xliff to automatically keep track of that sort of thing.

I've confirmed that this works fine in a few languages, but not all.

**Pull request merged**